### PR TITLE
Lottie icons 2408 (Milestone 1)

### PR DESCRIPTION
**Pull request by ohrytskov**
#2408 

Replace Toolbar SVGs with New Static Icons (Milestone 1)

### CODE REVIEW

File: src/components/RedoIcon.tsx
Comment (raineorshine) [inline src/components/RedoIcon.tsx:20]: This -1px marginTop might be affecting the positioning across font sizes since it does not scale. You can adjust the font size using the links in the footer:
image

I tested the vertical centering by scrolling the toolbar a bit to the right (and I temporarily deleted the padding on the scroll arrow for closer examination).

## This PR

Font Size 18 (default):
Brave Browser 2024-09-28 17 56 02

Font Size 25:
Brave Browser 2024-09-28 17 55 18

# `main`

While the icons are admittedly slightly lower than center, the difference is not amplified when the font size increases.

Font Size 18 (default):
Brave Browser 2024-09-28 17 59 34

Font Size 25:
Brave Browser 2024-09-28 17 59 48

Note that I'm not sure if it is the toolbar button icons or the arrow icon is wrong, but either way they need to be aligned vertically along their centers regardless of font size.

Let me know if that makes sense!

File: src/components/icons/IndentIcon.tsx
Comment (raineorshine) [inline src/components/icons/IndentIcon.tsx:5]: I'm not sure why `react-refresh/only-export-components` is needed here, since it is a component. does it need to be typed as `FC`?

File: src/components/RedoIcon.tsx
Comment (raineorshine) [inline src/components/RedoIcon.tsx:8]: If the scaling factor is the same for all icons, we should probably add it to `constants.ts` rather than repeating it each time.

File: src/components/RedoIcon.tsx
Comment (raineorshine) [inline src/components/RedoIcon.tsx:7]: I think changing the default font size to 14 is going to break some of the icons that are used in multiple places, such as https://github.com/cybersemics/em/blob/main/src/components/icons/HomeIcon.tsx. The size should correspond to font size, which defaults to 18. Can you change the default size to 18 and the scale factor to 1.066, or am I missing something?

Can you change the default font size to 18 and the scale factor to 1.066, or does that mess things up?

**Comment by raineorshine** [comment]
Note that the puppeteer snapshot tests are expected to fail because the toolbar appearance is changing. You can regenerate all the snapshots at the end by running `yarn test:puppeteer -u`.

File: src/components/RedoIcon.tsx
Comment (ohrytskov) [reply src/components/RedoIcon.tsx:20]: The new icons are centered within the SVG and include some padding, while the previous icons were aligned to the top-left corner. I aimed to position the new icons at the same level as the previous ones for consistency. Scaling with state.fontSize is part of Milestone 3, so for now, we're focused on achieving perfect positioning without scaling. A good option for now would be to remove the margin-top: -1px entirely, as all the icons will be aligned on the same level after replacement.

Comment (raineorshine) [reply src/components/RedoIcon.tsx:20]: Great! We'll revisit in Milestone 3.

File: src/components/icons/IndentIcon.tsx
Comment (ohrytskov) [reply src/components/icons/IndentIcon.tsx:5]: The eslint-disable comment was carried over from another file where it was needed, but it's redundant in this component and can be removed.

File: src/components/RedoIcon.tsx
Comment (ohrytskov) [reply src/components/RedoIcon.tsx:8]: Based on the current set of replaced SVGs, the scaling factor is indeed the same for all icons, though this wasn't clear before the replacements, and it’s still unknown for other icons. We can move the scaling factor to constants.ts, and if a different factor is needed for a specific icon later on, it can be overridden by defining a new one instead of importing from constants. So yes, let's move it to constants.

File: src/components/RedoIcon.tsx
Comment (ohrytskov) [reply src/components/RedoIcon.tsx:7]: Scaling is a part of Milestone 3, so we can keep all default values of size for now.

Factors to Keep in Mind for Milestone 3:

Size can be set by passing the parameter when rendering the icon component.
The default font size is set to 18 in constants, and it will remain as 18, while each icon component has its own default size prop.
The HomeIcon currently doesn’t have a default size.
Undo and redo are used not only in the toolbar but also in the modal window. We can pass size={14} explicitly from the modal window component for these icons.

Comment (raineorshine) [reply src/components/RedoIcon.tsx:7]: Thanks! We can revisit in Milestone 3.

File: src/components/icons/OutdentIcon.tsx
Comment (trevinhofmann) [inline src/components/icons/OutdentIcon.tsx:5]: The `react-refresh/only-export-components` rule can be re-enabled safely, and we should be able to remove `jsdoc/require-jsdoc` by adding a simple JSDoc comment here.

File: src/components/icons/Sort.tsx
Comment (trevinhofmann) [inline src/components/icons/Sort.tsx:170]:
```suggestion
/** Sort Icon Component. */
```

File: src/components/icons/StarIcon.tsx
Comment (trevinhofmann) [inline src/components/icons/StarIcon.tsx:7]: This should be `/** Favorite Icon Component. */` rather than left blank.

**Review by trevinhofmann — CHANGES_REQUESTED** [review]
Thank you, @ohrytskov! I've left a few initial comments. Also, the snapshots still need to be updated with `yarn test:puppeteer -u`.

File: src/components/icons/subCategorizeOneIcon.tsx
Comment (raineorshine) [reply src/components/icons/subCategorizeOneIcon.tsx:9]: Yes, I'm open to that. Let's wait till after Milestone 2 of #2408 so that we can see the full extent of the common icon functionality including static styles and animation logic.

File: src/components/icons/DeleteIcon.tsx
Comment (raineorshine) [reply src/components/icons/DeleteIcon.tsx:6]: Yes, we'll have to normalize these sizes, but it's okay to leave it to Milestone 3 as long as the visual is correct now.

File: src/components/icons/IndentIcon.tsx
Comment (raineorshine) [reply src/components/icons/IndentIcon.tsx:8]: Ideally all the icons will be scaled the same and will default to size 18. Plan on addressing that in Milestone 3.

File: src/components/RedoIcon.tsx
Comment (ohrytskov) [reply src/components/RedoIcon.tsx]: Scaling factor has been moved to constants.

**Comment by trevinhofmann** [comment]
Hi, @ohrytskov. I see you've added commits since my last review. When the PR is ready again for review, please just assign it back to me and/or let me know in the comments. Thank you!

**Comment by ohrytskov** [comment]
Hi @trevinhofmann, thank you for your initial feedback. I've made the updates to the PR. Could you please take another look?

**Comment by raineorshine** [comment]
@ohrytskov Could you update the puppeteer snapshot tests, too?

**Comment by ohrytskov** [comment]
@raineorshine Thanks for the reminder! I ran yarn test:puppeteer -u, and the snapshots are all updated now.

**Review by trevinhofmann — CHANGES_REQUESTED** [review]
Thank you, @ohrytskov! There are a few merge conflicts with the Puppeteer snapshots which will need to be resolved.

**Comment by ohrytskov** [comment]
Hi @trevinhofmann,

In this update:

1. Renamed the scaling factor and updated its description for clarity.
2. Added ConvertToNoteIcon for the Convert to Note action, while keeping PencilIcon for the Note action.

Please review and let me know if any changes are needed.

Thanks!

**Review by trevinhofmann — CHANGES_REQUESTED** [review]
Thank you very much for making those changes, @ohrytskov! Could you please resolve the merge conflicts before I give this a final glance?

**Comment by ohrytskov** [comment]
The merge conflicts have been resolved, and the branch is now ready for final review.

**Comment by raineorshine** [comment]
Thanks. Please pause work on this until #2432 is merged. Expecting significant merge conflicts as previously discussed.

**Comment by ohrytskov** [comment]
Got it, @raineorshine. I’ll pause work on this branch until #2432 is merged. I’ll keep an eye out for potential merge conflicts once that’s done. Thanks for the update!

**Comment by raineorshine** [comment]
Several snapshot tests seem to be failing, but I'm not clear why. Browserless should get rid of platform differences, so the same snapshots are generated regardless of your machine.

- font-size-22-one-thought-1-diff - React error I haven't seen before in a test.
- font-size-22-initial-load-1-diff - Related to the Swap Parent toolbar icon.
- drag-drop-hover-after-table-column-one-1-diff - Appears to be an alert fade-in timing issue (Animations will be disabled in tests soon: #2110)

**Comment by raineorshine** [comment]
I'm going to let @trevinhofmann review and help you fix the snapshot tests.

**Review by raineorshine — CHANGES_REQUESTED** [review]
Looks good, thanks!

Regarding the timeout comments, I'd like to explain where I'm coming from. I understand the the tests can be flaky sometimes, which is an annoying problem. We are continually trying to make the tests more stable across platforms. `docker` and `browserless` have helped a lot in ensuring tests are run the same and snapshots match across platforms. Different machines will still run tests at different speeds, which can sometimes affect test outcomes if the timeouts are too short.

That said, we need to understand the implications of increasing sleeps in the tests. This slows down tests for *every developer that works on the project*. It has a major impact on DX which should not be underestimated. This is why I created #2110, #2111, and #2054. We're currently in the middle of a big effort to speed up the puppeteer tests. This is especially important as the Puppeteer tests are the most realistic integration tests. They are very important for catching real-world usage and browser-specific behaviors, and they will only grow in number.

Increasing a delay from 100ms to 1000ms may seem like a small change, but it adds up and affects the whole team. For that reason I will be defending the current timeouts and requiring substantial explanation and justification for increasing them. If it's absolutely necessary, that I'm totally okay with that. But we need to prove that it is absolutely necessary and have good documentation for it given the repercussions.

Hope that makes sense, and I have communicated the need without coming across as too heavy-handed :).

**Comment by raineorshine** [comment]
You probably noticed that the tests were refactored today... twice! They are stable now. The new technique is to import the `page` object directly from setup. Let me know if you need any help.

File: src/e2e/puppeteer/__tests__/drag-and-drop.ts
Comment (ohrytskov) [reply src/e2e/puppeteer/__tests__/drag-and-drop.ts:245]: The 1000ms delay in the drag-and-drop tests was introduced solely for debugging purposes.

File: src/shortcuts/archive.tsx
Comment (ohrytskov) [reply src/shortcuts/archive.tsx]: Comment removed as suggested.

File: src/e2e/puppeteer/helpers/waitForEditable.ts
Comment (ohrytskov) [reply src/e2e/puppeteer/helpers/waitForEditable.ts]: The increased timeout was added temporarily for debugging purposes to investigate a specific issue. I don't anticipate needing more than 6 seconds in normal scenarios.

File: src/e2e/puppeteer/__tests__/ContextView.ts
Comment (ohrytskov) [reply src/e2e/puppeteer/__tests__/ContextView.ts:37]: The reference to the Settings button was part of the debugging process to figure out why the test wasn't working. The main idea behind using scrollIntoView was to ensure the Context View button was visible and clickable. Without scrolling, buttons outside the viewport can't be clicked, due to security and optimization limitations.

Comment (ohrytskov) [reply src/e2e/puppeteer/__tests__/ContextView.ts]: Scrolling directly to the Context View button left it only partially visible, causing click failures. To address this, initial scrolling was adjusted to the Settings button to better center the Context View. Following optimization suggestions, the scroll now targets the next sibling, the Note button, for improved visibility and click success.

File: src/e2e/puppeteer/helpers/scrollIntoView.ts
Comment (ohrytskov) [reply src/e2e/puppeteer/helpers/scrollIntoView.ts]: The scrolling behavior has been updated to 'instant' to improve test speed. Additionally, the 1-second sleep delay (used after scrolling and before the Context View click) has been removed, while the other two sleep calls have been reverted to their original durations. These changes support our ongoing effort to minimize animations in Puppeteer tests.

File: src/e2e/puppeteer/helpers/scrollIntoView.ts
Comment (ohrytskov) [reply src/e2e/puppeteer/helpers/scrollIntoView.ts]: Got it! And thank you for pointing this out. I have modified the function to throw an explicit error if the element isn't found, ensuring that any absence is caught early in the testing process.

File: src/e2e/puppeteer/__tests__/drag-and-drop.ts
Comment (trevinhofmann) [reply src/e2e/puppeteer/__tests__/drag-and-drop.ts]: Safely removed in https://github.com/cybersemics/em/pull/2429/commits/88322dc8adb2d78e113f4654a672a1956dc8a264

File: src/e2e/puppeteer/__tests__/ContextView.ts
Comment (trevinhofmann) [reply src/e2e/puppeteer/__tests__/ContextView.ts]: This explanation made sense to me, but I went ahead and tested to verify. Switching from `Note` to `Context View` does indeed create an issue with clicking the icon.

* https://github.com/cybersemics/em/pull/2429/commits/8956f24bd3268892471e1c2c6a68224924d3f081
* https://github.com/cybersemics/em/pull/2429/commits/c9d9165bb4666c1bf1587f14e4bdccd8a4dd0aa4

**Review by trevinhofmann — APPROVED** [review]
Thank you for your patience with merge conflicts and test flakiness, @ohrytskov!

All of the new timeouts/sleeps seem to have been cleaned up, @raineorshine, and tests are passing reliably (in the CI; I have occasional flakiness on my own machine, unrelated to these changes).

This PR looks good to me at this point 🎉

File: src/e2e/puppeteer/__tests__/ContextView.ts
Comment (raineorshine) [inline src/e2e/puppeteer/__tests__/ContextView.ts:37]: Scrolling to the Note button is no better than scrolling to Settings button. You are still tightly coupling the Context View test to an unrelated button. What happens if the buttons change position in the Toolbar? Then this test will break. Instead you need to:

1. Scroll to the Context View button plus a few pixels.
2. Scroll to the "next" button after the Context View regardless of what that button is (using DOM traversal).

File: src/components/RedoIcon.tsx
Comment (raineorshine) [inline src/components/RedoIcon.tsx:20]: Why set `width` and `height` in the `style` attribute when they are already set in the width and height attributes? It seems like it should be either one or the other.

File: src/shortcuts/archive.tsx
Comment (raineorshine) [inline src/shortcuts/archive.tsx:45]: Not sure why these comments were deleted?

File: src/components/icons/splitSentencesIcon.tsx
Comment (raineorshine) [inline src/components/icons/splitSentencesIcon.tsx:8]: Component names should be PascalCase

File: src/components/icons/subCategorizeAllIcon.tsx
Comment (raineorshine) [inline src/components/icons/subCategorizeAllIcon.tsx:8]: Component names should be PascalCase

File: src/components/icons/subCategorizeOneIcon.tsx
Comment (raineorshine) [inline src/components/icons/subCategorizeOneIcon.tsx:8]: Component names should be PascalCase

File: src/shortcuts/home.tsx
Comment (raineorshine) [inline src/shortcuts/home.tsx:5]: At some point `HomeIcon` and `HomeToolbarIcon` will need to be combined.

File: src/components/icons/SettingsIcon.tsx
Comment (raineorshine) [inline src/components/icons/SettingsIcon.tsx:7]: Let's just document this component as "Settings icon."

File: src/components/icons/ShareIcon.tsx
Comment (raineorshine) [inline src/components/icons/ShareIcon.tsx:7]: Let's just document this component as "Share icon."

File: src/components/icons/Sort.tsx
Comment (raineorshine) [inline src/components/icons/Sort.tsx:15]: Remove "refactored"

File: src/components/icons/Sort.tsx
Comment (raineorshine) [inline src/components/icons/Sort.tsx:92]: Remove "refactored"

File: src/components/icons/StarIcon.tsx
Comment (raineorshine) [inline src/components/icons/StarIcon.tsx:7]: Remove "refactored"

File: src/components/icons/StarIcon.tsx
Comment (raineorshine) [inline src/components/icons/StarIcon.tsx:8]: The file name should always match the component name, so rename this file to `FavoritesIcon.tsx`.

File: src/components/icons/StrikethroughIcon.tsx
Comment (raineorshine) [inline src/components/icons/StrikethroughIcon.tsx:7]: Remove "refactored"

File: src/components/icons/SwapParentIcon.tsx
Comment (raineorshine) [inline src/components/icons/SwapParentIcon.tsx:7]: Remove "refactored"

File: src/components/icons/TextColor.tsx
Comment (raineorshine) [inline src/components/icons/TextColor.tsx:7]: Remove "refactored"

File: src/components/icons/UnderlineIcon.tsx
Comment (raineorshine) [inline src/components/icons/UnderlineIcon.tsx:7]: Remove "refactored"

**Review by raineorshine — CHANGES_REQUESTED** [review]
Think of JSDOC comments as permanent documentation for a function or variable. They are good for describing what a function does, why it does it, where it's used, and any edge cases that the developer should know about. This is why I recommend removing "refactored" from the JSDOC's. It's not pertinent information for future developers. We always have the commit history to see how a piece of code has been refactored over time.

I noticed a positioning issue with the icons that are used in the QuickDrop components:

## Steps to Reproduce

Drag a thought.

## Current Behavior

The Copy to Clipboard icon smaller than the other icons and is not centered.

Brave Browser 2024-10-15 08 35 43

## Expected Behavior

QuickDrop icons should be the same size and centered.

Brave Browser 2024-10-15 08 48 25

File: src/e2e/puppeteer/helpers/waitForEditable.ts
Comment (raineorshine) [reply src/e2e/puppeteer/helpers/waitForEditable.ts]: Got it! Sounds good.

Just make sure to add a comment on the PR for changes that you are still testing, or switch the PR to a draft. Otherwise we will assume that it is ready for review.

**Comment by ohrytskov** [comment]
Recent Updates Ready for Review:

SVG Update: Removed width and height from top-level SVG tags.
New Helper: Added scrollBy in ContextView test.
Renaming: Changed StarIcon.tsx to FavoritesIcon.tsx.
Consistency: Standardized icon component names and enhanced the Copy Clipboard icon.
Docs: Updated comments in icons and archive shortcut.

**Review by raineorshine — APPROVED** [review]

**Pull request merged**
